test/CadlRanchProjects.Tests/type-property-nullable.cs

```diff
@@ -285,5 +289,175 @@ public Task Type_Property_Nullable_CollectionsString_patchNull() => Test(async (
+            Assert.AreEqual(new[] { "a", "b" }, model.NullableProperty);
+        }
+
+        [Test]
+        public void RequiredNullableCollectionsSerializeAsNull()
+        {
+            var inputModel = new CollectionsStringProperty("required", new List<string> { "1", "2" });
+
+            Assert.NotNull(inputModel.NullableProperty);
+
+            inputModel.NullableProperty = null;
+
+            var element = JsonAsserts.AssertWireSerializes(inputModel);
+
+            Assert.AreEqual(JsonValueKind.Null, element.GetProperty("nullableProperty").ValueKind);
+        }
+
+        [Test]
+        public void CollectionPropertiesCanBeMutatedAfterConstruction()
+        {
+            var inputModel = new CollectionsStringProperty(
+                "string",
+                Array.Empty<string>()
+            );
+
+            inputModel.NullableProperty.Add("1");
+
+            Assert.AreEqual(1, inputModel.NullableProperty.Count);
+        }
+
+        [TestCase("J")]
+        [TestCase("W")]
+        public void ValidateWritenullableModel(string format)
+        {
+            var options = format switch
+            {
+                "W" => new ModelReaderWriterOptions("W"),
+                "J" => ModelReaderWriterOptions.Json,
+                _ => throw new NotSupportedException()
+            };
+            var output = TypePropertyNullableModelFactory.StringProperty(requiredProperty: "requiredProperty");
+
+            var result = ModelReaderWriter.Write(output, options);
+            using var document = JsonDocument.Parse(result);
+
+            foreach (var property in document.RootElement.EnumerateObject())
+            {
+                if (property.NameEquals("requiredProperty"))
+                {
+                    Assert.AreEqual("requiredProperty", property.Value.GetString());
+                    continue;
+                }
+            }
+        }
+
+        [TestCase("J")]
+        [TestCase("W")]
+        public void ValidateReadInputModel(string format)
+        {
+            var options = format switch
+            {
+                "W" => new ModelReaderWriterOptions("W"),
+                "J" => ModelReaderWriterOptions.Json,
+                _ => throw new NotSupportedException()
+            };
+            var json = @"{""requiredProperty"": ""foo""}";
+            var binary = BinaryData.FromString(json);
+            var model = ModelReaderWriter.Read<StringProperty>(binary, options);
+
+            Assert.AreEqual("foo", model.RequiredProperty);
+            if (format == "J")
+            {
+                var field = typeof(StringProperty).GetField("_serializedAdditionalRawData", BindingFlags.Instance | BindingFlags.NonPublic);
+                var rawData = (IDictionary<string, BinaryData>)field.GetValue(model);
+                Assert.AreEqual(0, rawData.Count);
+            }
+        }
```

Comment:
we have other places for tests like this, therefore here we could remove these two test cases.

Reply:
removed